### PR TITLE
test: component integration testing w/ mock redux store

### PR DESCRIPTION
**Pull request by 121watts**
Closes #

_Briefly describe your proposed changes:_

  - [x] Rebased/mergeable
  - [x] Tests pass
  - [x] http/swagger.yml updated (if modified Go structs or API)
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: ui/src/clockface/components/dropdowns/Dropdown.tsx [hunk 1]
```diff
@@ -43,6 +43,8 @@ export interface Props {
   mode?: DropdownMode
   titleText?: string
   menuHeader?: JSX.Element
+  testid: string
```
Comment (chnn) [inline ui/src/clockface/components/dropdowns/Dropdown.tsx]:
```suggestion
  testID: string
```

😈

File: ui/src/clockface/components/dropdowns/Dropdown.tsx [hunk 2]
```diff
@@ -43,6 +43,8 @@ export interface Props {
...
+  buttonTestId: string
```
Comment (chnn) [inline ui/src/clockface/components/dropdowns/Dropdown.tsx]:
```suggestion
  buttonTestID: string
```

File: ui/src/clockface/components/dropdowns/Dropdown.tsx [hunk 3]
```diff
@@ -174,7 +179,10 @@ class Dropdown extends Component<Props, State> {
             autoHeight={true}
             maxHeight={maxMenuHeight}
           >
-            <div className="dropdown--menu">
+            <div
+              className="dropdown--menu"
+              data-testid={`dropdown--menu ${testid}`}
```
Comment (chnn) [inline ui/src/clockface/components/dropdowns/Dropdown.tsx]: should this just be?

```suggestion
              data-testid={testid}
```

File: ui/src/shared/components/TimeMachineQueryBuilder.test.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+import {waitForElement, fireEvent} from 'react-testing-library'
+
+import TimeMachineQueryBuilder from 'src/shared/components/TimeMachineQueryBuilder'
+
+jest.mock('src/shared/apis/v2/queryBuilder')
+
+let setInitialState
+describe('TimeMachineQueryBuilder', () => {
+  beforeEach(() => {
+    setInitialState = state => {
```
Comment (chnn) [inline ui/src/shared/components/TimeMachineQueryBuilder.test.tsx]: This probably doesn't need to live in the `beforeEach` hook

File: ui/src/shared/components/TimeMachineQueryBuilder.test.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    }
+  })
+
+  it('can select a bucket', async () => {
+    const {getByTestId, getAllByTestId, queryAllByTestId} = renderWithRedux(
+      <TimeMachineQueryBuilder />,
+      setInitialState
+    )
+
+    let bucketsDropdownClosed = await waitForElement(() =>
```
Comment (chnn) [inline ui/src/shared/components/TimeMachineQueryBuilder.test.tsx]: so great

File: ui/src/shared/components/TimeMachineQueryBuilder.test.tsx [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+      getByTestId('buckets--button')
+    )
+
+    fireEvent.click(bucketsDropdownClosed)
+
+    const bucketItems = getAllByTestId(/dropdown--item/)
+
+    expect(bucketItems.length).toBe(2)
+
+    const b2 = getByTestId('dropdown--item b2')
```
Comment (chnn) [inline ui/src/shared/components/TimeMachineQueryBuilder.test.tsx]: I wonder if we need to use the `getByTestId` helper so liberally. It seems to be described as a "last resort" sort of selector by react-testing-library if there's nothing else we can use.

File: ui/src/shared/reducers/v2/timeMachines.ts [hunk 7]
```diff
@@ -667,7 +678,10 @@ const setViewProperties = (
   const view: any = state.view
   const properties = view.properties
 
-  return {...state, view: {...view, properties: {...properties, ...update}}}
+  return {
+    ...state,
+    view: {...view, properties: {...properties, ...update}},
```
Comment (chnn) [inline ui/src/shared/reducers/v2/timeMachines.ts]: is this one a prettier config mismatch??

File: ui/src/shared/components/TimeMachineQueryBuilder.test.tsx (hunk 4 above)
Comment (ischolten) [reply ui/src/shared/components/TimeMachineQueryBuilder.test.tsx]: + 1

**Pull request merged**